cloudpickle/cloudpickle.py

```diff
@@ -934,7 +936,6 @@ def subimport(name):
 def dynamic_subimport(name, vars):
     mod = imp.new_module(name)
     mod.__dict__.update(vars)
-    sys.modules[name] = mod
```

Comment:
This should not be changed in this PR, right? This line come from a very old commit (e7341b67).

The fact that this deleted line does not break any existing test is worrisome but this should not be changed by this PR in my opinion.

Reply:
Actually, registering dynamic modules was a bug. Once registered as such, subsequent calls to dumps will consider the module as not dynamic which will make it impossible to load in other python processes.